### PR TITLE
Updating oecert to take a custom data file.

### PR DESCRIPTION
**Pull request by jazzybluesea**
Signed-off-by: Sergio Wong <sergio.wong@gmail.com>

### CODE REVIEW

File: tests/tools/oecert/host/host.cpp [hunk 1]
```diff
@@ -147,17 +150,58 @@ static oe_result_t _gen_cert(
...
+        fclose(fp);
+
+        result = get_remote_report_with_custom_data(
+            enclave,
+            &ecall_result,
+            buffer,
+            size,
+            (uint8_t**)&remote_report,
+            &report_size);
+
```
Comment (yentsanglee) [inline tests/tools/oecert/host/host.cpp:198]: buffer needs to be freed.

File: tests/tools/oecert/enc/enc.cpp [hunk 2]
```diff
@@ -112,6 +113,37 @@ oe_result_t get_tls_cert_signed_with_key(
...
+    uint8_t** report_buffer,
+    size_t* report_buffer_size)
+{
+    oe_result_t result = OE_FAILURE;
+
+    uint8_t sha256[32];
+
+    init_mbedtls();
+
+    if (Sha256(custom_data, custom_data_size, sha256) != 0)
```
Comment (yentsanglee) [inline tests/tools/oecert/enc/enc.cpp:128]: Hash of custom_data is embedded in the report but the custom_data itself is not carried over. What is the user scenario? Cannot the custom_data_file itself be the hash value?

File: tests/tools/oecert/oecert.edl [hunk 3]
```diff
@@ -19,5 +19,11 @@ enclave {
             size_t public_key_size,
             [out] unsigned char **data, 
             [out] size_t *data_size);
+
+        public oe_result_t get_remote_report_with_custom_data(
```
Comment (dthaler) [inline tests/tools/oecert/oecert.edl:23]: the term "report" is legacy, is this really needed or can we use "evidence" instead?

File: tests/tools/oecert/host/host.cpp (hunk 1 above)
Comment (jazzybluesea) [reply tests/tools/oecert/host/host.cpp:198]: Lines 308-309 will free remote_report?

File: tests/tools/oecert/enc/enc.cpp (hunk 2 above)
Comment (jazzybluesea) [reply tests/tools/oecert/enc/enc.cpp:128]: I wanted to generate a report with the sha256 of the custom data.  The custom data plus the report is then sent in the body for request to MAA.  I just needed a tool to do this for me.

File: tests/tools/oecert/enc/enc.cpp [hunk 4]
```diff
@@ -112,6 +113,37 @@ oe_result_t get_tls_cert_signed_with_key(
...
+    {
+        goto done;
+    }
+
+    result = oe_get_report(
```
[the remark this answers is not in the document]
Comment (jazzybluesea) [reply tests/tools/oecert/enc/enc.cpp:133]: I see what you are saying, maybe for a different PR, my goals was to update the tool to support what I needed. A lot of the existing code and samples will need to migrated to use the new API...

File: tests/tools/oecert/enc/enc.cpp (hunk 2 above)
Comment (yentsanglee) [reply tests/tools/oecert/enc/enc.cpp:128]: MAA takes evidence type, evidence, and enclave held data. In your case, evidence type is oe_report and enclave help data is the hash of a public key. So your custom_data is the public key, right?

**Comment by jazzybluesea** [comment]
Closing for now.

**Pull request closed without merging**